src/loaders.rs

```diff
@@ -31,6 +32,45 @@ fn safe_join(directory: &Path, template_name: &str) -> Option<PathBuf> {
     }
 }
 
+#[cached(
+    size = 128,          // Cache size
+    result = true,       // Cache Result type
+    key = "String",      // Use owned String as key
+    convert = r##"{ dirname.to_string() }"## // Convert &str to String
+)]
+fn get_app_template_dirs(py: Python<'_>, dirname: &str) -> Result<Vec<String>, PyErr> {
+    // Import django.apps
+    let apps_module = PyModule::import(py, "django.apps")?;
+
+    // Get the 'apps' object
+    let apps = apps_module.getattr("apps")?;
+
+    // Call get_app_configs() on apps
+    let app_configs = apps.call_method0("get_app_configs")?;
+
+    // Convert to iterator and process each app_config
```

Comment:
I don't think we need the comments here:

```suggestion
    let apps_module = PyModule::import(py, "django.apps")?;
    let apps = apps_module.getattr("apps")?;
    let app_configs = apps.call_method0("get_app_configs")?;
```